mach_nix/data/providers.py

```diff
@@ -503,10 +503,13 @@ def all_candidates(self, pkg_name, extras=None, build=None) -> Iterable[Candidat
 def conda_virtual_packages():
 
     packages = dict(
-        __glibc=environ.get("MACHNIX_GLIBC_VERSION", platform.libc_ver()[0][1]),
         __unix=0,
     )
 
+    libc_complier, libc_version = platform.libc_ver()
+    if libc_complier == 'glibc':
+        packages['__glibc'] = environ.get("MACHNIX_GLIBC_VERSION", libc_version),
+
```

Comment:
Thanks for fixing. Would you mind adding a comment in the code pointing out why this is required? (needed for macos, etc.)

Reply:
Excellent idea! I'll do that now!